arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java

```diff
@@ -753,7 +753,7 @@ public void tracePostExecution(MessageFrame frame, Operation.OperationResult ope
     }
 
     if (frame.getDepth() == 0
-        && (frame.getState() == MessageFrame.State.EXCEPTIONAL_HALT || opCode() == REVERT)) {
+        && (isExceptional() || opCode() == REVERT)) {
```

Comment:
It's unclear exactly when Besu's `MessageFrame` enters the `EXCEPTIONAL_HALT` state. This is particularly true wrt deployment related exceptions (`invalidCodePrefixException` and `maxCodeSizeException`) for whatever reason.